onnxruntime/core/framework/partial_executor.cc

```diff
@@ -0,0 +1,497 @@
+  // Calculate total input sizes for this operation.
+  input_activation_sizes = 0;
+  input_parameter_sizes = 0;
+  ORT_UNUSED_PARAMETER(node_name);
+  const int input_count = op_kernel_context->InputCount();
+  for (auto i = 0; i < input_count; i++) {
+    const OrtValue* p_input = op_kernel_context->GetInputMLValue(i);
+    if (p_input != nullptr && p_input->IsTensor()) {
+      const OpKernelInfo& op_kernel_info = p_op_kernel->Info();
+      const Tensor* p_tensor = nullptr;
+      bool is_param = op_kernel_info.TryGetConstantInput(i, &p_tensor);
+      if (!is_param) {
+        p_tensor = &(p_input->Get<Tensor>());
+      }
+      size_t tensor_size = p_tensor->SizeInBytes();
+
+#if defined(TRACE_EXECUTION)
+      const TensorShape& tensor_shape = p_tensor->Shape();
+      size_t element_size = p_tensor->DataType()->Size();
+      LOGS(logger, INFO) << node_name << " input[" << i << "]"
+                         << " is_param=" << is_param
+                         << " size=" << tensor_size
+                         << " shape=" << tensor_shape.ToString()
+                         << " element_size=" << element_size
+                         << "\n";
+#endif
+      if (is_param) {
+        input_parameter_sizes += tensor_size;
+      } else {
+        input_activation_sizes += tensor_size;
+      }
+    }
+  }
+}
+
+static Status ReleaseNodeMLValues(ExecutionFrame& frame,
+                                  const SequentialExecutionPlan& seq_exec_plan,
+                                  const SequentialExecutionPlan::NodeExecutionPlan& node_exec_plan,
+                                  const logging::Logger& logger) {
+  for (auto i = node_exec_plan.free_from_index; i <= node_exec_plan.free_to_index; ++i) {
+    auto ort_value_idx = seq_exec_plan.to_be_freed[i];
+    VLOGS(logger, 1) << "Releasing ort_value with index: " << ort_value_idx;
+    ORT_RETURN_IF_ERROR(frame.ReleaseMLValue(ort_value_idx));
+  }
+
+  return Status::OK();
+}
+
+Status PartialExecutor::Execute(const SessionState& session_state, const std::vector<int>& feed_mlvalue_idxs,
+                                const std::vector<OrtValue>& feeds, const std::vector<int>& fetch_mlvalue_idxs,
+                                std::vector<OrtValue>& fetches,
+                                const std::unordered_map<size_t, CustomAllocator>& fetch_allocators,
+                                const logging::Logger& logger) {
+  const bool is_profiler_enabled = session_state.Profiler().IsEnabled();
+  TimePoint tp;
+  TimePoint sync_time_begin;
+  TimePoint kernel_begin_time;
+  size_t input_activation_sizes = 0;
+  size_t input_parameter_sizes = 0;
+  size_t total_output_sizes = 0;
+
+  if (is_profiler_enabled) {
+    tp = session_state.Profiler().Now();
+  }
+
+  if (state_.GetExecutionFrame() == nullptr) {
+    state_.GetExecutionFrame() = onnxruntime::make_unique<ExecutionFrame>(feed_mlvalue_idxs, feeds, fetch_mlvalue_idxs, fetches, fetch_allocators, session_state);
```

Comment:
nit: keep lines to 120 chars.

[https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelines](https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelines) is very helpful for this

Reply:
Thanks, I just found out VS Code has a setting for this that adds a ruler, "editor.rulers": [120], so nice, see below snapshot :)

![image](https://user-images.githubusercontent.com/1211949/115676260-418eb400-a304-11eb-8fbc-d4e64be76ac6.png)